src/main/java/org/generationcp/middleware/api/nametype/GermplasmNameTypeServiceImpl.java

```diff
@@ -126,4 +140,35 @@ public long countAllNameTypes() {
 			.collect(Collectors.toList());
 	}
 
+	@Override
+	public boolean isNameTypeUsedInListDataProp(final String nameType) {
+		return this.daoFactory.getListDataPropertyDAO().isNameTypeInUse(nameType);
+	}
+
+	@Override
+	public void updateNameType(final Integer nameTypeId, final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO) {
+		final UserDefinedField userDefinedField = this.daoFactory.getUserDefinedFieldDAO().getById(nameTypeId);
+
+		if (!StringUtils.isBlank(germplasmNameTypeRequestDTO.getCode())) {
+			userDefinedField.setFcode(germplasmNameTypeRequestDTO.getCode());
+		}
+		if (!StringUtils.isBlank(germplasmNameTypeRequestDTO.getName())) {
+			userDefinedField.setFname(germplasmNameTypeRequestDTO.getName());
+		}
+		if (!StringUtils.isBlank(germplasmNameTypeRequestDTO.getDescription())) {
+			userDefinedField.setFdesc(germplasmNameTypeRequestDTO.getDescription());
+		}
+
+		userDefinedField.setFuid(ContextHolder.getLoggedInUserId());
```

Comment:
Does the fuid represent the created by user? in that case fuid should not be modified when updating
Same logic with fdate. 
If everything is blank, it does not make sense to edit the name type, it can be controlled in BMSAPI. Usually when the request body in a patch is empty (everything is null) NO_CONTENT is returned 
See GermplasmResource.updateGermplasmBasicDetails

Reply:
done!